pkg/config/app_config.go

```diff
@@ -156,6 +157,11 @@ func loadUserConfig(configFiles []string, base *UserConfig) (*UserConfig, error)
 			return nil, err
 		}
 
+		content, err = migrateUserConfig(path, content)
```

Comment:
I would like a way to track which 'version' of the config we're on so that we only need to run migrations if we are behind. This has a few benefits:
* No need to enforce idempotency in migrations
* Less frequently messing with the user's config (I suspect that due to indentation, changes the file will often be overwritten despite no semantic change)
* Less work on startup (though I suspect it's not currently eating up much time).

We could store this version on the config itself. What do you think?

Reply:
I don't think this would work. If we read a config file that doesn't have a version, there's no way to tell whether this is a new user who just created a new config file for the latest version, or if it's an old config file from before we started to put a version in the file. Unless we _require_ the user to manually add the correct version when they initially create a config file, which I don't think we should.

I also disagree with some of the alleged downsides:
- _Less frequently messing with the user's config_: on the contrary, if we put a version number in the file, we'll have to rewrite the file for _every_ user whenever we add a new migration. If we don't version the file, we only need to rewrite it for those users who have the migrated key in their config. This requires adding a commit to the underlying PR that avoids rewriting the file when nothing changed; see https://github.com/jesseduffield/lazygit/pull/2718#issuecomment-1605417507.
- _Less work on startup_: That's true, but I agree that it's probably not too significant.
- So that leaves the idempotence question. I don't feel this is going to be a problem for the kind of migration that we are likely to add: renaming a key, or changing the type of a key (e.g. from a bool to an enum). Can you think of a migration that would be problematic in this regard?

Reply:
Good points. Let's go with your approach then :)